bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/rest/PersistenceResource.java

```diff
@@ -0,0 +1,261 @@
+
+        final Object responseObject = getItemHistoryBean(serviceName, itemName, startTime, endTime);
+        return Response.ok(responseObject).build();
+    }
+
+    private Date convertTime(String sTime) {
+        DateTimeType dateTime = new DateTimeType(sTime);
+        return dateTime.getCalendar().getTime();
+    }
+
+    private ItemHistoryBean getItemHistoryBean(String serviceName, String itemName, String timeBegin, String timeEnd) {
+        PersistenceService service = persistenceServices.get(serviceName);
+
+        long timerStart = System.currentTimeMillis();
+
+        if (service == null) {
+            logger.debug("Persistence service not found '{}'.", serviceName);
+            throw new WebApplicationException(404);
+        }
+
+        if (!(service instanceof QueryablePersistenceService)) {
+            logger.debug("Persistence service not queryable '{}'.", serviceName);
+            throw new WebApplicationException(404);
+        }
+
+        QueryablePersistenceService qService = (QueryablePersistenceService) service;
+
+        Date dateTimeBegin = new Date();
+        Date dateTimeEnd = dateTimeBegin;
+        if (timeBegin != null) {
+            dateTimeBegin = convertTime(timeBegin);
+        }
+
+        if (timeEnd != null) {
+            dateTimeEnd = convertTime(timeEnd);
+        }
+
+        // End now...
+        if (dateTimeEnd.getTime() == 0) {
+            dateTimeEnd = new Date();
+        }
+        if (dateTimeBegin.getTime() == 0) {
+            dateTimeBegin = new Date(dateTimeEnd.getTime() - 86400000);
+        }
+
+        // Default to 1 days data if the times are the same
+        if (dateTimeBegin.getTime() >= dateTimeEnd.getTime()) {
+            dateTimeBegin = new Date(dateTimeEnd.getTime() - 86400000);
+        }
+
+        FilterCriteria filter = new FilterCriteria();
+        Iterable<HistoricItem> result;
+        State state = null;
+
+        Long quantity = 0l;
+        double average = 0;
+        Double minimum = null;
+        Double maximum = null;
+        Date timeMinimum = null;
+        Date timeMaximum = null;
+
+        ItemHistoryBean bean = null;
+        bean = new ItemHistoryBean();
+
+        bean.name = itemName;
+
+        // First, get the value at the start time.
+        // This is necessary for values that don't change often otherwise data will start after the start of the graph
+        // (or not at all if there's no change during the graph period)
+        filter = new FilterCriteria();
+        filter.setEndDate(dateTimeBegin);
+        filter.setItemName(itemName);
+        filter.setPageSize(1);
+        filter.setOrdering(Ordering.DESCENDING);
+        result = qService.query(filter);
+        if (result != null && result.iterator().hasNext()) {
+            HistoricItem historicItem = result.iterator().next();
+
+            double value = bean.addData(dateTimeBegin.getTime(), historicItem.getState());
```

Comment:
Why do you use the unprecise double here instead of continuing to work with BigDecimal, which is used internally in the state?